R-package/vignettes/CharRnnModel.Rmd

```diff
@@ -0,0 +1,277 @@
+Char RNN Example
+=============================================
+
+This example aims to show how to use lstm model to build a char level language model, and generate text from it. We use a tiny shakespeare text for demo purpose.
+
+Data can be found at https://github.com/dmlc/web-data/tree/master/mxnet/tinyshakespeare. 
+
+Preface
+-------
+This tutorial is written in Rmarkdown.
+- You can directly view the hosted version of the tutorial from [MXNet R Document](http://mxnet.readthedocs.org/en/latest/package/r/CharRnnModel.html)
+- You can find the download the Rmarkdown source from [here](https://github.com/dmlc/mxnet/blob/master/R-package/vignettes/CharRnnModel.Rmd)
+
+Load Data 
+---------
+First of all, load in the data and preprocess it.
+```{r}
+require(mxnet)
+```
+Set basic network parameters.
+```{r}
+batch.size = 32
+seq.len = 32
+num.hidden = 256
+num.embed = 256
+num.lstm.layer = 2
+num.round = 5
+learning.rate= 0.1
+wd=0.00001
+clip_gradient=1
+update.period = 1
+```
+download the data.
+```{r}
+download.data <- function(data_dir) {
+    dir.create(data_dir, showWarnings = FALSE)
+    setwd(data_dir)
+    if (!file.exists('input.txt')) {
+        download.file(url='https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/tinyshakespeare/input.txt',
+                      destfile='input.txt', method='wget')
+    }
+    setwd("..")
```

Comment:
Can you change this a little bit? Please don't change the working directory, or it might fail other tests.